packages/blockchain-wallet-v4-frontend/src/scenes/TaxCenter/selectors.ts

```diff
@@ -0,0 +1,64 @@
+import { filter, lift, not, startsWith } from 'ramda'
+
+import { createDeepEqualSelector } from '@core/utils'
+import { selectors } from 'data'
+
+import { FIRST_YEAR } from './model'
+
+type BtcContext = {
+  addresses: Array<string>
+  bech32: Array<string>
+  legacy: Array<string>
+}
+
+export const getReportYearOptions = () => {
+  const CURRENT_YEAR = new Date().getFullYear()
+
+  const availableYears = CURRENT_YEAR - FIRST_YEAR
+  const options = [...new Array(availableYears)].map((_, i) => ({
+    text: `${FIRST_YEAR + i}`,
+    value: FIRST_YEAR + i
+  }))
+
+  // current year
+  options.push({
+    text: `${CURRENT_YEAR}`,
+    value: CURRENT_YEAR
+  })
+
+  // all time option
+  options.push({
+    text: `All Time`,
+    value: 0
+  })
+
+  return options
+}
+
+export const getAddressesAndXpubs = createDeepEqualSelector(
+  [
+    selectors.core.kvStore.eth.getDefaultAddress,
+    selectors.core.kvStore.xlm.getDefaultAccountId,
+    (state) => state
+  ],
+  (ethAccountR, xlmAccountR, state) => {
+    const transform = (ethAccount, xlmAccount) => {
+      // @ts-ignore
```

Comment:
need this?

Reply:
Yes, to allow the closure with the state